packages/product-exporter/src/main.js

```diff
@@ -0,0 +1,161 @@
+/* @flow */
+import type {
+  ApiConfigOptions,
+  ExportConfigOptions,
+  LoggerOptions,
+  ProductProjection,
+  ProcessFnResponse,
+} from 'types/product'
+import type {
+  Client,
+  ClientRequest,
+} from 'types/sdk'
+
+import { createClient } from '@commercetools/sdk-client'
+import { createRequestBuilder } from '@commercetools/api-request-builder'
+import { createHttpMiddleware } from '@commercetools/sdk-middleware-http'
+import {
+  createAuthMiddlewareForClientCredentialsFlow,
+} from '@commercetools/sdk-middleware-auth'
+import {
+  createUserAgentMiddleware,
+} from '@commercetools/sdk-middleware-user-agent'
+import JSONStream from 'JSONStream'
+import pkg from '../package.json'
+
+export default class ProductExporter {
+  // Set flowtype annotations
+  accessToken: string;
+  apiConfig: ApiConfigOptions;
+  client: Client;
+  exportConfig: ExportConfigOptions;
+  logger: LoggerOptions;
+
+  constructor (
+    apiConfig: ApiConfigOptions,
+    exportConfig: ExportConfigOptions,
+    logger: LoggerOptions,
+    accessToken: string,
+  ) {
+    this.apiConfig = apiConfig
+    this.client = createClient({
+      middlewares: [
+        createAuthMiddlewareForClientCredentialsFlow(this.apiConfig),
+        createUserAgentMiddleware({
+          libraryName: pkg.name,
+          libraryVersion: pkg.version,
+        }),
+        createHttpMiddleware({ host: this.apiConfig.apiUrl }),
+      ],
+    })
+
+    const defaultConfig = { staged: false, json: true }
+
+    this.exportConfig = { ...defaultConfig, ...exportConfig }
+    this.logger = {
+      error: () => {},
+      warn: () => {},
+      info: () => {},
+      debug: () => {},
+      ...logger,
+    }
+    this.accessToken = accessToken
+  }
+
+  run (outputStream: stream$Writable): Promise<*> {
+    this.logger.debug('Starting Export')
+    const formattedStream = ProductExporter._decideStream(
+      this.exportConfig.exportType,
+    )
+    this.logger.debug('Preparing outputStream')
+    formattedStream.pipe(outputStream)
+    return this._getProducts(formattedStream)
+    .catch((e: Error) => {
+      this.logger.debug(e, 'Oops. Something went wrong')
```

Comment:
I think the log level for this should be `error` since we are logging error. Any reason why you used `debug`?